### PR TITLE
some improvements for test coverage and fix for HTTP_HOST validation

### PR DESCRIPTION
**Pull request by boltnev**
Improvements for test coverage.
Also i made the fix for HTTP_HOST validation to make hosts with digits on the end of hostname(but not consisting with only digits) valid

### CODE REVIEW

File: iktomi/web/url.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import six
 if six.PY2:
     from urlparse import urlparse, parse_qs, unquote
-else:
+else:# pragma: no cover; we checking coverage only in python2 part
```
Comment (Harut) [inline iktomi/web/url.py]: english syntax error: правильно we check :)

File: iktomi/web/app.py [hunk 2]
```diff
@@ -16,15 +16,17 @@
 ip_number = '(\d|[1-9]\d|1\d{2}|2[0-4]\d|25[0-5])'
 dns_letter = '[a-z\d]([a-z\d\-]*[a-z\d])?'
 port = '(?::\d+)'
-HOSTNAME_REGEX = re.compile("^({letter}\.)*{letter}[a-z]{port}?$".format(letter=dns_letter, 
+HOSTNAME_REGEX = re.compile("^({letter}\.)*{letter}{port}?$".format(letter=dns_letter,
                                                                          port=port), re.I)
-IP_REGEX = re.compile("^({number}\.){times}{number}{port}?$".format(number=ip_number, 
-                                                                    times="{3}", 
+IP_REGEX = re.compile("^({number}\.){times}{number}{port}?$".format(number=ip_number,
```
Comment (ods) [inline iktomi/web/app.py]: Defining number pattern separately is quite reasonable to me. I'm ok with defining port separately too. But why times? Why not just write `{{3}}` in the first place?

Comment (ods) [inline iktomi/web/app.py]: Defining number pattern separately is quite reasonable to me. I'm ok with defining port separately too. But why times? Why not just write `{{3}}` in the first place?

File: iktomi/web/app.py [hunk 3]
```diff
@@ -16,15 +16,17 @@
...
+                                                                    times='{3}',
```
Comment (ods) [inline iktomi/web/app.py]: Defining number pattern separately is quite reasonable to me. I'm ok with defining port separately too. But why times? Why not just write `{{3}}` in the first place?

File: iktomi/web/app.py (hunk 2 above)
Comment (ods) [inline iktomi/web/app.py]: Defining number pattern separately is quite reasonable to me. I'm ok with defining port separately too. But why times? Why not just write `{{3}}` in the first place?

**Pull request merged**